Jucer2Reprojucer/main.cpp

```diff
@@ -270,6 +270,106 @@ void writeUserNotes(LineWriter& wLn, const juce::ValueTree& valueTree)
   }
 }
 
+// clang-format off
+/** Sanitizes a string to match the regex ^[A-Za-z0-9_]*$. Sanitization follows these rules: 
+  * 1) Invalid characters are replaced with "_".
+  * 2) At most one consecutive replacement "_" is kept (if the original name contained 
+  *    multiple "_" they remain unaltered).
+  * 3) No replacement "_" is added at the end or at the beginning.
+  */
+// clang-format on
+juce::String sanitizeName(const juce::String input)
+{
+  const juce::String validCharacters =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
+  juce::String output;
+  bool mustInsertReplacementChar = false;
+  for (int i = 0; i < input.length(); i++)
+  {
+    auto c = input[i];
+    // keep valid characters
+    if (validCharacters.containsChar(c))
+    {
+      if (mustInsertReplacementChar)
+      {
+        output += '_';
+        mustInsertReplacementChar = false;
+      }
+      output += c;
+      continue;
+    }
+    else
+    {
+      // don't add "_" at the beginning
+      if (output.length() == 0)
+        continue;
+      else
+      {
+        mustInsertReplacementChar = true;
+      }
+    }
+  }
+  return output;
+}
+
+// clang-format off
+/** Reads a jucerProject and sanitizes configuration names. Valid config names match the
+  * regex ^[A-Za-z0-9_]*$. Sanitization is done with sanitizeName(). An optional digit is 
+  * added until the configuration name is unique. The sanitization is synchronized across 
+  * all exporters in the project.
+  */
+// clang-format on
+void sanitizeConfigurationNames(const juce::ValueTree& jucerProject)
```

Comment:
This function is pretty well written, but it doesn't fit the architecture of `Jucer2Reprojucer`. More concretely, `Jucer2Reprojucer`:
- never modifies the original `ValueTree`,
- extracts data from the `ValueTree` right before transforming it and writing it to the `CMakeLists.txt` file, and
- tries to read the content of the `ValueTree` only once.

Could you please try to take the content of this function and move it to where we are already looping over `EXPORTFORMATS` and `CONFIGURATIONS`? This would guarantee that we are iterating in the same way when sanitizing and when outputting to the `CMakeLists.txt` file. It would also allow us to write the original configuration name as a comment when the sanitized name is different:
```cmake
jucer_export_target_configuration(
  "Visual Studio 2017"
  NAME "Debug_32_bit"  # "Debug (32-bit)" in MyGreatProject.jucer
  DEBUG_MODE ON
)
```
Finally, the warnings would be properly interleaved with other warnings that we might add in the future.